bin/propolis-server/src/lib/vm/state_driver.rs

```diff
@@ -893,6 +905,11 @@ mod tests {
         vcpu_ctrl.expect_pause_all().times(1).returning(|| ());
         vcpu_ctrl.expect_resume_all().times(1).returning(|| ());
 
+        // VMM will be paused once prior to exporting state, and then resumed
+        // afterwards when the migration fails.
+        vm_ctrl.expect_pause_vm().times(1).returning(|| ());
+        vm_ctrl.expect_resume_vm().times(1).returning(|| ());
```

Comment:
It might be worthwhile to add a `mockall::Sequence` or two in here to enforce that the `pause_vm`/`resume_vm` calls are ordered correctly with respect to the calls to deal with entities and vCPUs. (Arguably this test could always have had one to make sure we pause things before resuming them, but I think the value's much higher when we need to verify ordering amongst several pause or resume operations.)

Reply:
Good idea.  I've added the sequences and checked that the tests still pass.